src/android/KnoxPluginEnabled.kt

```diff
@@ -1,127 +1,314 @@
+                Timber.e("Reboot with Knox Manage API Response Error", error);
+            }) {
+            override fun getHeaders(): MutableMap<String, String> {
+                val headers = HashMap<String, String>()
+                headers["cache-control"] = "no-cache"
+                headers["content-type"] = "application/x-www-form-urlencoded"
+                headers["Authorization"] = "bearer $bearerToken"
+                return headers
+            }
+
+            // DeviceID != IMEI
+            override fun getParams(): MutableMap<String, String> {
+                val params = HashMap<String, String>()
+                params["deviceId"] = deviceId
+                return params
+            }
+        }
+
+        requestQueue.add(stringRequest)
+    }
+
+    private fun startRebootAlarm() {
+        Timber.d("startRebootAlarm()")
+
+        val context = cordova.context
+        val currentTime = System.currentTimeMillis()
+        val triggerTime = currentTime + TIME_SPAN_24_HOURS_MS
+        val alarmManager = context.getSystemService(ALARM_SERVICE) as AlarmManager
+        val alarmIntent = Intent(context, RebootTimeoutReceiver::class.java)
+
+        val pendingIntent = PendingIntent.getBroadcast(
+            context,
+            0,
+            alarmIntent,
+            PendingIntent.FLAG_IMMUTABLE
+        )
+
+        try {
+            alarmManager.setExact(AlarmManager.RTC_WAKEUP, triggerTime, pendingIntent)
+            Timber.d("will reboot at %s (from current time %s)", triggerTime, currentTime)
+        } catch (ex: SecurityException) {
+            Timber.e(ex, "failed to set reboot alarm due to security exception")
+        } catch (ex: Exception) {
+            Timber.e(ex, "failed to set reboot alarm due to unknown exception")
+        }
+    }
+
+    private fun authenticateAPI() {
+        try {
+            val context = cordova.context
+            Timber.d("Authenticating Knox Manage API")
+            val url =
+                "https://us02.manage.samsungknox.com/emm/oauth/token?grant_type=client_credentials&client_id=healthrecoverysolutions@development.healthrecoverysolutions.com&client_secret=HRSistheBest123!"
+            val requestQueue = Volley.newRequestQueue(context)
+            val stringRequest = object : StringRequest(
+                Method.POST, url,
+                Response.Listener { response ->
+                    Timber.v("Authenticating Knox Manage API response: $response")
+                    try {
+                        val responseObject = JSONObject(response)
+                        bearerToken = responseObject["access_token"].toString()
+
+                        // Getting device ID and passing along authentication
+                        if (deviceId == "") {
+                            getDeviceId()
+                        }
+                    } catch (e: JSONException) {
+                        e.printStackTrace()
```

Comment:
We should also send the exception to timber

`Timber.e(e, "knox manage authentication failed")`